tools/querytee/proxy_endpoint_test.go

```diff
@@ -350,6 +350,97 @@ func Test_ProxyEndpoint_Comparison(t *testing.T) {
+	}{
+		"responses are below threshold": {
+			slowResponseThreshold:    100 * time.Millisecond,
+			preferredResponseLatency: 0 * time.Millisecond,
+			secondaryResponseLatency: 0 * time.Millisecond,
+		},
+		"one response above threshold": {
+			slowResponseThreshold:    100 * time.Millisecond,
+			preferredResponseLatency: 0 * time.Millisecond,
+			secondaryResponseLatency: 101 * time.Millisecond,
+		},
+		"responses are both above threshold, but lower than threshold between themselves": {
+			slowResponseThreshold:    100 * time.Millisecond,
+			preferredResponseLatency: 101 * time.Millisecond,
+			secondaryResponseLatency: 150 * time.Millisecond,
+		},
+		"responses are both above threshold, and above threshold between themselves": {
+			slowResponseThreshold:    100 * time.Millisecond,
+			preferredResponseLatency: 101 * time.Millisecond,
+			secondaryResponseLatency: 202 * time.Millisecond,
+		},
+	}
+
+	for name, scenario := range scenarios {
+		t.Run(name, func(t *testing.T) {
+			preferredBackend := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				w.Header().Set("Content-Type", "application/json")
+				w.WriteHeader(200)
+				time.Sleep(scenario.preferredResponseLatency)
+				_, err := w.Write([]byte("preferred response"))
+				require.NoError(t, err)
+			}))
+
+			defer preferredBackend.Close()
+			preferredBackendURL, err := url.Parse(preferredBackend.URL)
+			require.NoError(t, err)
+
+			secondaryBackend := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				w.Header().Set("Content-Type", "application/json")
+				w.WriteHeader(200)
+				time.Sleep(scenario.secondaryResponseLatency)
+				_, err := w.Write([]byte("preferred response"))
+				require.NoError(t, err)
+			}))
+
+			defer secondaryBackend.Close()
+			secondaryBackendURL, err := url.Parse(secondaryBackend.URL)
+			require.NoError(t, err)
+
+			backends := []*ProxyBackend{
+				NewProxyBackend("preferred-backend", preferredBackendURL, time.Second, true, false),
+				NewProxyBackend("secondary-backend", secondaryBackendURL, time.Second, false, false),
+			}
+
+			logger := newMockLogger()
+			reg := prometheus.NewPedanticRegistry()
+			comparator := &mockComparator{
+				comparisonResult: ComparisonSuccess,
+			}
+
+			endpoint := NewProxyEndpoint(backends, "test", NewProxyMetrics(reg), logger, comparator, scenario.slowResponseThreshold)
+
+			resp := httptest.NewRecorder()
+			req, err := http.NewRequest("GET", "http://test/api/v1/test", nil)
+			require.NoError(t, err)
+			endpoint.ServeHTTP(resp, req)
+
+			// The HTTP request above will return as soon as the primary response is received, but this doesn't guarantee that the response comparison has been completed.
+			// Wait for the response comparison to complete before checking the logged messages.
+			waitForResponseComparisonMetric(t, reg, ComparisonSuccess)
+
+			for _, m := range logger.messages {
+				fmt.Println(m)
```

Comment:
Is it intentional to use `fmt.Println` here, or is it debugging code?
```suggestion
				t.Log(m)
```

Reply:
Left over from debugging. Will remove